### ISSUE TITLE
Add configuration management and CLI

### ISSUE THREAD

**Issue by CooperBigFoot** [CooperBigFoot/RL-demo#10]
## Description
Implement configuration management for easy experimentation and command-line interface for training/evaluation.

## Reference
See planning document: `/scratchpads/phase3_integration_plan.md` - Section 4

## Dependencies
- #7 (Training script to be configurable)

## Tasks
- [ ] Create `src/rl_demo/configs/default_config.py`:
  - Environment parameters (reservoir capacities, etc.)
  - Network architecture settings
  - Training hyperparameters
  - Logging configuration
- [ ] Add CLI arguments to training script:
  - `--config`: Path to config file
  - `--exp-name`: Experiment name
  - `--device`: CPU/GPU selection
  - `--seed`: Random seed
  - `--resume`: Resume from checkpoint
- [ ] Support YAML config files:
  - Create `configs/experiments/baseline.yaml`
  - Allow config overrides from CLI
- [ ] Add experiment tracking:
  - Auto-generate experiment IDs
  - Save config with checkpoints
  - Track git commit hash

## Code Structure
```python
from dataclasses import dataclass
import argparse

@dataclass
class DQNConfig:
    # Environment
    v_max: float = 1000.0
    v_min: float = 100.0
    
    # Network
    hidden_dim: int = 128
    n_layers: int = 2
    
    # Training
    lr: float = 1e-3
    batch_size: int = 32
    buffer_size: int = 10000
    
    # Experiment
    exp_name: str = "dqn_reservoir"
    seed: int = 42

def parse_args():
    parser = argparse.ArgumentParser()
    parser.add_argument("--config", type=str, default="configs/default.yaml")
    parser.add_argument("--exp-name", type=str)
    parser.add_argument("--device", type=str, default="cuda")
    return parser.parse_args()
```

## Acceptance Criteria
- Easy to run experiments with different configs
- Config changes are tracked
- Can reproduce experiments from config
- CLI is intuitive and well-documented

## Priority
Low - Nice to have for experimentation
